wfe/web-front-end.go

```diff
@@ -169,27 +183,29 @@ func (wfe *WebFrontEndImpl) HandleFunc(mux *http.ServeMux, pattern string, h fun
 		if err == nil {
 			response.Header().Set("Replay-Nonce", nonce)
 		}
-		response.Header().Set("Access-Control-Allow-Origin", "*")
 
-		// Return a bodyless response to HEAD for any resource that allows GET.
-		if _, ok := methodsOK["GET"]; ok && request.Method == "HEAD" {
-			// We'll be sending an error anyway, but we
-			// should still comply with HTTP spec by not
+		switch request.Method {
+		case "HEAD":
```

Comment:
Does it make sense to also change the method to "GET" here before calling the handler? I think that would solve the other issue you mentioned, where HEAD to a challenge URL doesn't actually run getChallenge.